Sync v2-4-stable with v2-4-test to release 2.4.0

ephraimbuddy opened the pull request.

potiuk committed:
Fix case when SHELL variable is not set in kubernetes tests (#26235)

When SHELL variable is not set, kubernetes tests will fall back
to using 'bash'

(cherry picked from commit b7db311336bbe909ee7f559af2b1da7635c54823)

potiuk committed:
Make breeze works with latest docker-compose (#26233)

The latest docker-compose dropped an alias for `docker-compose` and
we need to detect it and use "docker compose" instead.

(cherry picked from commit 7f47006effb330429fb510eb52644d22544f4fad)

potiuk committed:
Limit eager upgrade of protobuf library to < 4.21.0 (#26243)

* Limit eager upgrade of protobuf library to < 4.21.0

Until all the Google client libraries get upgraded to >= 2.0.0, we need to
limit the protobuf version.

(cherry picked from commit c6053f52908d6f1605b5e9a0566a653e90cc2d0e)

jedcunningham committed:
Reformat 2.4.0 release notes (#26247)

This makes them proper sections and reorders a couple of them as well.

(cherry picked from commit 49fb3ffb99705dda77f0b57c9d6be7829e4549f5)

potiuk committed:
Skip overriding version from parameter with the one from tag (#26244)

Since we are not always updating the version suffix in code
(rc1/rc2/b1/b2), the tag specified via --tag-build prefix should
override the one in code rathe than the other way round.

So what's left now - we will just print warning if the suffix does
not match.

(cherry picked from commit 9cf6f6a63d923e01fc7a3a634a0fa75f43c4c198)

potiuk committed:
Add better progress in CI for constraints generation. (#26253)

Currently constraints generation is not really showing good progress
while the packages are being removed/installed. This adds progress
that shows that something happens.

(cherry picked from commit d6f473b31d9902290dc3a204657b4a7ed8d7843b)

potiuk committed:
The PROD cache is pushed always in regular cache build step (#26254)

In v2-4-test it turned out that PROD cache build was a bit
too limited - the cache has not been prepared if branch was not
main (copy&paste victim). This PR fixes this - cache is always build
in merge run regardless of the branch we are in.

(cherry picked from commit 41785457d29154b7ee626713b1994fb2dc444615)

edithturn committed:
D400 first line should end with period batch02 (#25268)

(cherry picked from commit 50668445137e4037bb4a3b652bec22e53d1eddd7)

cra committed:
Add more weekday operator and sensor examples #26071 (#26098)

(cherry picked from commit dd6b2e4e6cb89d9eea2f3db790cb003a2e89aeff)

uranusjr committed:
Fix 'from airflow import version' lazy import (#26239)

This incorrectly accesses the version *attribute* in airflow.version
instead of the airflow.version module itself, breaking compatibility.

I actually rewrote the entire __getattr__ hook; much of the logic seems
to be quite redundant.

(cherry picked from commit b7a603cf89728e02a187409c83983d58cc554457)

ashb committed:
Update erd SHA to pass static checks (#26263)

The cause of the failure was a PR race -- the PR that changed migrations
was opened and passing tests before the ERD sha was added.

(cherry picked from commit 12b544502a55540476267859c8bc558195a7588e)

DavidKatz-il committed:
Add the dag_id to AirflowDagCycleException message (#26204)

(cherry picked from commit aada79e1dabe729362ddd8a3c7eb43efb8fc1c17)

dstandish committed:
Don't blow up when a task produces a dataset that is not consumed. (#26257)

If you have a dataset outlet on a task, and no DAG was recorded as
consuming that dataset it failed with a null value constraint violation
in the db

(cherry picked from commit fe82e9609522fa302d3eaf7193689718571d4af4)

potiuk committed:
Make "quick build" actually test build time (#26251)

* Make "quick build" actually test build time

There was a "string" instead of array passed in case of --max-time
command flag which caused an error - regardless of the time it
took to run the build, but it was also ignored because the
job was run within "continue-on-error" cache push job.

This PR fixes it to use proper "exit" command and separates it out
to a job where quick build failure will be noticed (not in PRs but
in the merge-run)

(cherry picked from commit 63da96b4331bdc70ec49e360e87610960b31371c)

jedcunningham committed:
Flush dataset events before queuing dagruns (#26276)

When we go to schedule dagruns from the dataset dagrun queue, we assume
the events will happen before the queue records, which isn't the case
unless we explicitly flush them first. This ensures that dagruns are
properly related to their upstream dataset events.

(cherry picked from commit 954349a952d929dc82087e4bb20d19736f84d381)

potiuk committed:
Properly build URL to retrieve logs independently from system (#26337)

The previous way of building the path depended on the OS path
but it was really used to build the URL so we should use
urllib instead of os.path.join.

(cherry picked from commit 18386026c28939fa6d91d198c5489c295a05dcd2)

jedcunningham committed:
Fix `dags_needing_dagruns` dataset info timestamp (#26288)

(cherry picked from commit 3c9c0f940b67c25285259541478ebb413b94a73a)

uranusjr committed:
Flag to deserialize value on custom XCom backend (#26343)

(cherry picked from commit ffee6bceb32eba159a7a25a4613d573884a6a58d)

jedcunningham committed:
Add more to the ignore list for non-core changes (#26355)

(cherry picked from commit b966b8d881849051e79692fd8bf7b090af29291c)

blag committed:
Fix Dataset bugs in grid view (#26356)

* Fix bugs in grid view

* Add DatasetModel.uri to group_by

(cherry picked from commit 9a9ed5b1f6c3caf1525eb73c4d1440752f737f5e)

ephraimbuddy committed:
Update Airflow version

ephraimbuddy committed:
Update Release Note for 2.4.0

potiuk committed:
Apply PEP-563 (Postponed Evaluation of Annotations) to core airflow (#26290)

This PR applies PEP-563 to all core airflow Python files - i.e.
those that we want to cherry-pick to v2-* branches in order
to make cherry-picking for the upcoming 2.4* releases easier. There was
a separate PR (#26289) with non-core changes that are not going to be
cherry-picked..

This PR is a result of combining some of the 47 PRs reviewed
and approved separately (otherwise it would have been unreviewable)

The history of those PRs can be changed in:
https://github.com/apache/airflow/pulls?q=is%3Apr+label%3Afuture-annotations+is%3Aopen

Relevant discussion: https://lists.apache.org/thread/81fr042s5d3v17v83bpo24tnrr2pp0fp
Lazy consensus call: https://lists.apache.org/thread/l74nvjh8tgbtojllhwkcn7f8mfnlz4jq

(cherry picked from commit d67ac5932dabbf06ae733fc57b48491a8029b8c2)

uranusjr committed:
Handle list when serializing expand_kwargs (#26369)

(cherry picked from commit b816a6b243d16da87ca00e443619c75e9f6f5816)

potiuk committed:
Move the deserialization of custom XCom Backend to 2.4.0 (#26392)

The change has been cherry-picked so we need to update the docs.

(cherry picked from commit 41e48aaf4dbec24c924a03198b224a0b81c432d4)

uranusjr committed:
Work around pyupgrade edge cases (#26384)

(cherry picked from commit 9444d9789bc88e1063d81d28e219446b2251c0e1)

dstandish committed:
Respect max_active_runs for dataset-triggered dags (#26348)

Co-authored-by: Ash Berlin-Taylor <ash_github@firemirror.com>
Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
(cherry picked from commit b99d1cd5d32aea5721c512d6052b6b7b3e0dfefb)

ephraimbuddy committed:
Fix pre-commit for checking revision heads map (#26373)

We need to check with the base_version instead of the parsed version obj

(cherry picked from commit af39faafb7fdd53adbe37964ba88a3814f431cd8)

ashb committed:
Better validation of Dataset URI during dag parse (#26389)

Previously we had the validation on the Dataset model, but we since
moved the "dag" facing class to a separate one. This adds validation to
the public class, and extends the validation to not allow space-only
strings

(cherry picked from commit bd181daced707680ed22f5fd74e1e13094f6b164)

jedcunningham committed:
Require dag_id arg for dags list-runs (#26357)

While it would ideal to transition to a positional arg like was
attempted in #25978, this unfortunately does result in a breaking change
so we cannot do it now.

By instead marking the existing arg as required, we maintain backcompat
while also providing a helpful error message to the user if they forget
it.

This reverts commit ed6ea72f181a1d381cc1ff6c801f10cc0bc0d830.

(cherry picked from commit f6c579c1c0efb8cdd2eaf905909cda7bc7314f88)

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Consolidate to one ``schedule`` param (#25410)
+- DAG regex flag in backfill command (#23870)
+- Adding support for owner links in the Dags view UI (#25280)
+- clear specific dag run TI (#23516)
```
```suggestion
- Ability to clear a specific DAG Run's task instances via REST API (#23516)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- DAG regex flag in backfill command (#23870)
+- Adding support for owner links in the Dags view UI (#25280)
+- clear specific dag run TI (#23516)
+- Possibility to document DAG with a separated markdown file (#25509)
```
```suggestion
- Possibility to document DAG with a separate markdown file (#25509)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Create new databases from the ORM (#24156)
+- Implement ``XComArg.zip(*xcom_args)`` (#25176)
+- Introduce ``sla_miss`` metric (#23402)
+- Migrate files to ts (#25267)
```
I'm not sure this should be a new feature. Maybe improvement or misc?

jedcunningham replied:
There is already a ts one in improvements, so either there or move both to misc. I lean toward misc.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Add default task retry delay config (#23861)
+- Add clear DagRun endpoint. (#23451)
+- Add support for timezone as string in cron interval timetable (#23279)
+- add auto refresh to dags home page (#22900)
```
```suggestion
- Add auto-refresh to dags home page (#22900)
```

Nit: the only non-cap one here!

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Remove mapped operator validation code (#25870)
+- More ``DAG(schedule=...)`` improvements (#25648)
+- Reduce ``operator_name`` dupe in serialised JSON (#25819)
+- make grid view group/mapped summary UI  more consistent (#25723)
```
```suggestion
- Make grid view group/mapped summary UI more consistent (#25723)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Reduce ``operator_name`` dupe in serialised JSON (#25819)
+- make grid view group/mapped summary UI  more consistent (#25723)
+- Remove useless statement in ``task_group_to_grid`` (#25654)
+- Add auto refresh to grid logs (#25621)
```
Should we wrap this into the grid logs line in new features? This is an improvement on a new feature.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- make grid view group/mapped summary UI  more consistent (#25723)
+- Remove useless statement in ``task_group_to_grid`` (#25654)
+- Add auto refresh to grid logs (#25621)
+- Grid Logs, add visual indicator of selected attempt (#25611)
```
Same here.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Add auto refresh to grid logs (#25621)
+- Grid Logs, add visual indicator of selected attempt (#25611)
+- Add optional data interval to ``CronTriggerTimetable`` (#25503)
+- Simply json responses (#25518)
```
This should probably be misc or skip, it's not user facing.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Add and document description fields (#25370)
+- Improve Airflow logging for operator Jinja template processing (#25452)
+- Update core example DAGs to use ``@task.branch`` decorator (#25242)
+- Fix DAG ``audit_log`` route (#25415)
```
```suggestion
- Update DAG ``audit_log`` route (#25415)
```

"Fix" makes me think bugfix (which it could also be, depends on your perspective I guess). I'm good with this change here though.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Improve Airflow logging for operator Jinja template processing (#25452)
+- Update core example DAGs to use ``@task.branch`` decorator (#25242)
+- Fix DAG ``audit_log`` route (#25415)
+- Add missing import in best-practices code example (#25391)
```
This should be docs only.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Remove ``getTasks`` from Grid view (#25359)
+- Improve taskflow type hints with ParamSpec (#25173)
+- Use tables in grid details panes (#25258)
+- Disallow any dag tags longer than 100 char (#25196)
```
Bugfix?

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Use tables in grid details panes (#25258)
+- Disallow any dag tags longer than 100 char (#25196)
+- Explicitly list ``@dag`` arguments (#25044)
+- Only refresh active dags on dags page (#24770)
```
Should this be wrapped into the auto-refresh new feature entry?

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- More typing in ``SchedulerJob`` and ``TaskInstance`` (#24912)
+- Patch ``getfqdn`` with more resilient version (#24981)
+- Replace all ``NBSP`` characters by whitespaces (#24797)
+- Have consistent types between the ORM and the migration files (#24044)
```
Bugfix?

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Rework contract of try_adopt_task_instances method (#23188)
+- Make ``expand()`` error vague so it's not misleading (#24018)
+- Add enum validation for ``[webserver]analytics_tool`` (#24032)
+- Split contributor's quick start into separate guides. (#23762)
```
```suggestion
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Add ``dttm`` searchable field in audit log (#23794)
+- Allow more parameters to be piped through via ``execute_in_subprocess`` (#23286)
+- Use ``func.count`` to count rows (#23657)
+- remove stale serialized dags (#22917)
```
```suggestion
- Remove stale serialized dags (#22917)
```

nit

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
-- ``ExternalTaskSensor`` now supports the ``soft_fail`` flag to skip if external task or DAG enters a failed state. (#23647)
+- Add the dag_id to ``AirflowDagCycleException`` message (#26204)
+- Properly build URL to retrieve logs independently from system (#26337)
+- Fix ``from airflow import version`` lazy import (#26239)
```
This should be bundled with the lazy import misc item.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Fix faulty executor config serialization logic (#26191)
+- Show ``DAGs`` and ``Datasets`` menu links based on role permission (#26183)
+- Allow setting ``TaskGroup`` tooltip via function docstring (#26028)
+- Rewrite recursion into iteration (#26175)
```
```suggestion
- Fix RecursionError on graph view of a DAG with many tasks (#26175)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Rewrite recursion into iteration (#26175)
+- Fix backfill occasional deadlocking (#26161)
+- Fix ``DagRun.start_date`` not set during backfill with ``--reset-dagruns`` True (#26135)
+- use label instead of id for dynamic task labels in graph (#26108)
```
```suggestion
- Use label instead of id for dynamic task labels in graph (#26108)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Fix legacy timetable schedule interval params (#25999)
+- Fix response schema for ``list-mapped-task-instance`` (#25965)
+- Properly check the existence of missing mapped TIs (#25788)
+- fix broken auto-refresh (#25950)
```
```suggestion
- Fix broken auto-refresh on grid view (#25950)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Filter XCOM by key when calculating map lengths (#24530)
+- Fix ``ExternalTaskSensor`` not working with dynamic task (#25215)
+- Added exception catching to send default email if template file raises any exception (#24943)
+- Use ``sql_alchemy_conn`` for celery result backend when ``result_backend`` is not set (#24496)
```
I feel like this should be an improvement.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+Misc/Internal
+^^^^^^^^^^^^^
+
+- Add automatically generated ``ERD`` schema for the ``MetaData`` DB (#26217)
```
docs only

ephraimbuddy replied:
This one seems like it's good to be in misc. I take doc only to mean that the files are all rst?

jedcunningham replied:
Keep in mind this is end user centric. The static check parts of this isn't relevant, the only user facing change is the addition to the docs. At least that's how I view it 🤷‍♂️.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Bump ``moment-timezone`` from ``0.5.34`` to ``0.5.35`` in ``/airflow/www`` (#26080)
+- Fix Flask deprecation warning (#25753)
+- Add ``CamelCase`` to generated operations types (#25887)
+- Fix migration issues and tighten the CI upgrade/downgrade test (#25869)
```
Bugfix?

ephraimbuddy replied:
Joined with #24044

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Remove Smart Sensors (#25507)
+- Fix ``elasticsearch`` test config to avoid warning on deprecated template (#25520)
+- Bump ``terser`` from ``4.8.0 to 4.8.1`` in /airflow/ui (#25178)
+- Refactor test DAGs to unclutter example DAGs (#25327)
```
This should probably be a skip or AIP-48.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Fix ``elasticsearch`` test config to avoid warning on deprecated template (#25520)
+- Bump ``terser`` from ``4.8.0 to 4.8.1`` in /airflow/ui (#25178)
+- Refactor test DAGs to unclutter example DAGs (#25327)
+- Add ts types generation to static checks (#25167)
```
This can probably be skipped. Definitely not user facing.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Upgrade grid Table component to ts. (#25074)
+- Skip mapping against mapped ``ti ``if it returns None (#25047)
+- Refactor ``js`` file structure (#25003)
+- Add absolute paths for ``typescript`` files (#24978)
```
I'm also not sure we need this type of change either, also not user facing. I don't have a strong opinion here though.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Remove "bad characters" from our codebase (#24841)
+- Remove ``xcom_push`` flag from ``BashOperator`` (#24824)
+- Move Flask hook registration to end of file (#24776)
+- Remove defaults from Airflow test config (#24263)
```
```suggestion
```

Not sure why this lives outside of tests... probably should move it.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Preserve original order of providers' connection extra fields in UI (#24425)
+- Rename ``charts.css`` to ``chart.css`` (#24531)
+- Rename ``grid.css`` to ``chart.css`` (#24529)
+- misc: create new process group by ``set_new_process_group`` utility (#24371)
```
```suggestion
- Misc: create new process group by ``set_new_process_group`` utility (#24371)
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Rename ``charts.css`` to ``chart.css`` (#24531)
+- Rename ``grid.css`` to ``chart.css`` (#24529)
+- misc: create new process group by ``set_new_process_group`` utility (#24371)
+- Remove limit of presto-python-client version (#24305)
```
Not exactly sure the history here, but this is a noop - it was already this was in 2-3-stable.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Improve docker documentation (#25735)
+- Correctly link to Dag parsing context in docs (#25722)
+- Add note on ``task_instance_mutation_hook`` usage (#25607)
+- Documentation update. Actualizing passing data options (#25577)
```
```suggestion
- Note that TaskFlow API automatically passes data between tasks (#25577)
```

Or something.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Update DAG run to clarify when a DAG actually runs (#25290)
+- Update tutorial docs to include a definition of operators (#25012)
+- Rewrite the Airflow documentation home page (#24795)
+- Fix docs build errors for Sphinx 5 (#24870)
```
```suggestion
```

I think we can skip this one.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -27,123 +27,354 @@ Airflow 2.4.0beta1 (2022-09-08)
+- Update tutorial docs to include a definition of operators (#25012)
+- Rewrite the Airflow documentation home page (#24795)
+- Fix docs build errors for Sphinx 5 (#24870)
+- update ``task-generated mapping`` example (#23424)
```
```suggestion
- Fix ``task-generated mapping`` example (#23424)
```

jedcunningham reviewed (commented):
We also need the AIP-48 catchall in new features.

jedcunningham committed:
Clear autoregistered DAGs if there are any import errors (#26398)

We need to clear any autoregistered DAGs that may have been already
registered if we encounter any import errors while parsing a given DAG
file.

This maintains the behavior before we autoregistered DAGs.

(cherry picked from commit 01e3fb7eac6cceb1fa4cc68ee1e4fe3682edbf0a)

jedcunningham committed:
Fix UI redirect (#26409)

Co-authored-by: Konstantin Weddige <konstantin.weddige@lutrasecurity.com>
(cherry picked from commit 56e7555c42f013f789a4b718676ff09b4a9d5135)

jedcunningham committed:
Add min attrs version (#26408)

I'm not sure when we started using attrs directly, but we do and we need
>=22.1.0 as we use the min_length validator.

(cherry picked from commit fdecf12051308a4e064f5e4bf5464ffc9b183dad)

ashb reviewed (commented):
Oh. We need to add datasets to the release notes!

potiuk commented on airflow/api_connexion/openapi/v1.yaml:
```diff
@@ -1412,6 +1412,23 @@ paths:
+
+            This parameter is not meaningful when using the default XCom backend.
+
+            *New in version 2.5.0*
```
```suggestion
            *New in version 2.4.0*
```

potiuk commented on airflow/www/static/js/types/api-generated.ts:
```diff
@@ -3413,6 +3413,20 @@ export interface operations {
+         *
+         * This parameter is not meaningful when using the default XCom backend.
+         *
+         * *New in version 2.5.0*
```
```suggestion
         * *New in version 2.4.0*
```

potiuk reviewed (approved):
Reviewed all, all looks familiar. I additionally cherry-picked one change #26392 which was missing (not sure where and how)

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,129 +21,401 @@
-  .. code-block:: python
+This allows DAG runs to be automatically created as a result of a task "producing" a dataset. (In some ways
+this can be thought of as the inverse of ``TriggerDagRunOperator``, where instead of the producing DAG
+controlling which DAGs get created, the consuming DAGs can "listen" for changes.)
```
```suggestion
This allows DAG runs to be automatically created as a result of a task "producing" a dataset. In some ways
this can be thought of as the inverse of ``TriggerDagRunOperator``, where instead of the producing DAG
controlling which DAGs get created, the consuming DAGs can "listen" for changes.
```

While I love my parenthesis 🙃, I don't think we need them here.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,129 +21,401 @@
-         ...
 
+    # Or for classic operators
+    BashOperator(task_id="update-ds", bash_command=..., outluets=[dataset])
```
```suggestion
    BashOperator(task_id="update-ds", bash_command=..., outlets=[dataset])
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,129 +21,401 @@
 
-     dag_maker()
+(If you have the producer and consumer in different files you do not need to use the same Dataset object, two
+``Dataset()``\s created with the same URI are equal.)
```
```suggestion
If you have the producer and consumer in different files you do not need to use the same Dataset object, two
``Dataset()``\s created with the same URI are equal.
```

Same here.

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,129 +21,401 @@
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
 
-- DAG runs sorting logic changed in grid view (#25410)
+Previously you had do assign a DAG to a module-level variable in order for Airflow to pick it up. For example this
```
```suggestion
Previously you had to assign a DAG to a module-level variable in order for Airflow to pick it up. For example this

```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,129 +21,401 @@
+``airflow.contrib`` packages and deprecated modules are dynamically generated (#26153, #26179, #26167)
+""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
+
+The ``airflow.contrib`` packages and deprecated modules from Airflow 1.10 in ``airflow.hooks``, ``airflow.operators``, ``airflow.sensors`` packages, have now dynamically generated modules and while users can continue using the deprecated contrib classes, they are no longer visible for static code check tools and will be reported as missing. It is recommended for the users to move to non-deprecate ...
```
```suggestion
The ``airflow.contrib`` packages and deprecated modules from Airflow 1.10 in ``airflow.hooks``, ``airflow.operators``, ``airflow.sensors`` packages are now dynamically generated modules and while users can continue using the deprecated contrib classes, they are no longer visible for static code check tools and will be reported as missing. It is recommended for the users to move to the non-deprecated classes.
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,129 +21,401 @@
+``DBApiHook`` and ``SQLSensor`` have moved (#24836)
+"""""""""""""""""""""""""""""""""""""""""""""""""""
+
+``DBApiHook``, ``SQLSensor`` have been moved to the ``apache-airflow-providers-common-sql`` provider.
```
```suggestion
``DBApiHook`` and ``SQLSensor`` have been moved to the ``apache-airflow-providers-common-sql`` provider.
```

jedcunningham commented on RELEASE_NOTES.rst:
```diff
@@ -21,129 +21,402 @@
+capability - in this release we are adding the foundational feature that we will build upon.
 
-  .. code-block:: python
+For more info on Datasets please see :doc:`/concepts/datasets`.
```
```suggestion
For more info on Datasets please see :doc:`/concepts/datasets`.

Expanded dynamic task mapping support
"""""""""""""""""""""""""""""""""""""

Dynamic task mapping now includes support for ``expand_kwargs``, ``zip`` and ``map``.

For more info on dynamic task mapping please see :doc:`/concepts/dynamic-task-mapping`.
```
